src/app/SimpleSubscriptionResumptionStorage.cpp

```diff
@@ -0,0 +1,473 @@
+    {
+        if (mStorage->SyncDoesKeyExist(DefaultStorageKeyAllocator::SubscriptionResumption(subscriptionIndex).KeyName()))
+        {
+            subscriptionCount++;
+        }
+    }
+
+    return subscriptionCount;
+}
+
+CHIP_ERROR SimpleSubscriptionResumptionStorage::Delete(uint16_t subscriptionIndex)
+{
+    return mStorage->SyncDeleteKeyValue(DefaultStorageKeyAllocator::SubscriptionResumption(subscriptionIndex).KeyName());
+}
+
+CHIP_ERROR SimpleSubscriptionResumptionStorage::Load(uint16_t subscriptionIndex, SubscriptionInfo & subscriptionInfo)
+{
+    Platform::ScopedMemoryBuffer<uint8_t> backingBuffer;
+    backingBuffer.Calloc(MaxSubscriptionSize());
+    ReturnErrorCodeIf(backingBuffer.Get() == nullptr, CHIP_ERROR_NO_MEMORY);
+
+    uint16_t len = static_cast<uint16_t>(MaxSubscriptionSize());
+    ReturnErrorOnFailure(mStorage->SyncGetKeyValue(DefaultStorageKeyAllocator::SubscriptionResumption(subscriptionIndex).KeyName(),
+                                                   backingBuffer.Get(), len));
+
+    TLV::ScopedBufferTLVReader reader(std::move(backingBuffer), len);
+
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag()));
+
+    TLV::TLVType subscriptionContainerType;
+    ReturnErrorOnFailure(reader.EnterContainer(subscriptionContainerType));
+
+    // Node ID
+    ReturnErrorOnFailure(reader.Next(kPeerNodeIdTag));
+    ReturnErrorOnFailure(reader.Get(subscriptionInfo.mNodeId));
+
+    // Fabric index
+    ReturnErrorOnFailure(reader.Next(kFabricIndexTag));
+    ReturnErrorOnFailure(reader.Get(subscriptionInfo.mFabricIndex));
+
+    // Subscription ID
+    ReturnErrorOnFailure(reader.Next(kSubscriptionIdTag));
+    ReturnErrorOnFailure(reader.Get(subscriptionInfo.mSubscriptionId));
+
+    // Min interval
+    ReturnErrorOnFailure(reader.Next(kMinIntervalTag));
+    ReturnErrorOnFailure(reader.Get(subscriptionInfo.mMinInterval));
+
+    // Max interval
+    ReturnErrorOnFailure(reader.Next(kMaxIntervalTag));
+    ReturnErrorOnFailure(reader.Get(subscriptionInfo.mMaxInterval));
+
+    // Fabric filtered boolean
+    ReturnErrorOnFailure(reader.Next(kFabricFilteredTag));
+    ReturnErrorOnFailure(reader.Get(subscriptionInfo.mFabricFiltered));
+
+    // Attribute Paths
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_List, kAttributePathsListTag));
+    TLV::TLVType attributesListType;
+    ReturnErrorOnFailure(reader.EnterContainer(attributesListType));
+
+    size_t pathCount = 0;
+    ReturnErrorOnFailure(reader.CountRemainingInContainer(&pathCount));
+
+    // If a stack struct is being reused to iterate, free the previous paths ScopedMemoryBuffer
+    subscriptionInfo.mAttributePaths.Free();
+    if (pathCount)
+    {
+        subscriptionInfo.mAttributePaths.Calloc(pathCount);
+        ReturnErrorCodeIf(subscriptionInfo.mAttributePaths.Get() == nullptr, CHIP_ERROR_NO_MEMORY);
```

Comment:
Again, could test either the return value of `Calloc`, or just `ReturnErrorCodeIf(!subscriptionInfo.mAttributePaths, CHIP_ERROR_NO_MEMORY);